_posts/2023-07-05-niklas-fuzzing.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "Niklas Gögge on Fuzz Testing"
+permalink: /blog/2023/07/05/fuzzing/
+layout: post
+author: brink
+name: "brink"
+alt: Niklas Gögge on Fuzz Testing
+category: "News"
+description: Niklas Gögge on Fuzz Testing.
+---
+
+In a security-focused project like Bitcoin Core, testing is critical. [Bitcoin
+Core unit tests][] are written in C++ using the Boost libraries to quickly test
+components of the codebase in isolation (see the [bech32 unit tests][] for
+example). [Bitcoin Core functional tests][] are written in python to test
+features of the software, involving multiple components working together (see
+the [P2P ping functional tests][]).
+
+A less-familiar type of testing is fuzz testing where software repeatedly calls
```

Comment:
Why "less-familiar" and not "more cutting-edge" or something?

I think this description can be a lot more compelling. Imo calling patterns are not as important as the fact that (1)Fuzzing is characterized by the fact that inputs are (hopefully intelligently) generated by the fuzzing engine (2) fuzzing scales wayyy better than human code review, you can often just throw more money at it to find more bugs.